GithubApiWithVIPER/GithubApiWithVIPERTests/GithubApiWithVIPERTests.swift

```diff
@@ -99,3 +115,22 @@ private class MockRouter: GithubSearchRouterInterface {
         repoNames.append(repoName)
     }
 }
+
+private class MockInteractor: GithubSearchInteractorInterface {
+    weak var output: GithubSearchInteractorOutput?
+    var searchKeys: [String] = []
+    
+    func search(with text: String) {
```

Comment:
Aslında burası mock class olduğu için bu method'un içinde bir şey return etmen gerekmiyor. Yani burasını yine bir bool değer veya 
**var searchCalledWithText: String?**

gibi bir variable ile test etmen lazım.

Asıl output ile çağrılan methodları zaten yukarıda test ettin

**test_HandleSearchResult_Failure_ShouldInvokeShowError**
**test_HandleSearchResult_Success_ShouldCheckDataSource**

yani bu mock class logic barındırmamalı sadece cağrıldı çağrılmadı veya şununla çağrıldıyı vermeli. Eğer Interactor'ın test'ini yazman gerekseydi orada mockOutput ile buraları test edebilirdin ama gerek yok şu an